FreeRTOS/FreeRTOS-Plus-TCP
# DriverSAM: let gmac_dev_read() return the proper length (version 2)

Description
-----------
This PR replaces #950 which was compared to the earlier branch "dev/IPv6_integration".

Test Steps
-----------
I sent (IPv6) pings of increasing size from 0 up to to 1452 bytes, where my MTU was 1500.

    ping -c 1 -s 0 fe80::8008
    ping -c 1 -s 1 fe80::8008
    ...
    ping -c 1 -s 1452 fe80::8008  // Results in a packet of 1500 + 14 bytes.

The longest packet for ICMPv4 echo is 1472 bytes.

Checklist:
----------




By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
